kubernetes/src/main/java/com/linecorp/armeria/client/kubernetes/ArmeriaWebSocket.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.kubernetes;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.websocket.CloseWebSocketFrame;
+import com.linecorp.armeria.common.websocket.WebSocketCloseStatus;
+import com.linecorp.armeria.common.websocket.WebSocketFrame;
+import com.linecorp.armeria.common.websocket.WebSocketWriter;
+
+import io.fabric8.kubernetes.client.http.WebSocket;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+
+final class ArmeriaWebSocket implements WebSocket, Subscriber<WebSocketFrame> {
+    private final WebSocketWriter writer;
+    private final AtomicLong pending = new AtomicLong();
+    private final WebSocket.Listener listener;
+
+    @Nullable
+    private Subscription inboundSubscription;
+
+    ArmeriaWebSocket(WebSocketWriter writer, WebSocket.Listener listener) {
+        this.writer = writer;
+        this.listener = listener;
+    }
+
+    @Override
+    public boolean send(ByteBuffer buffer) {
+        // 'buffer' may be mutated by the caller, so we need to copy it.
+        final ByteBufAllocator alloc = ClientRequestContext.mapCurrent(RequestContext::alloc,
+                                                                       () -> ByteBufAllocator.DEFAULT);
+        final ByteBuf data = alloc.buffer(buffer.remaining()).writeBytes(buffer.duplicate());
+        final int dataLength = data.readableBytes();
+        pending.addAndGet(dataLength);
+        final boolean success = writer.tryWrite(WebSocketFrame.ofPooledBinary(data));
```

Comment:
Question) I believe this piece of code will send a frame with the FIN bit set.

i.e. This is the equivalent of:
```
WebSocketFrame#ofPooledBinary(data, true)
```

Is this intentional?